### PR TITLE
CIS Tests for Symmetric and Asymmetric Keys

### PR DESCRIPTION
**Pull request by tboggiano**
# A New PR

THANK YOU - We love to get PR's and really appreciate your time and help to improve this module

## Accepting a PR

Before we accept the PR - please confirm that you have run the tests locally to avoid our automated build and release process failing. You can see how to do that in our wiki

https://github.com/sqlcollaborative/dbachecks/wiki 

## Please confirm you have 0 failing Pester Tests

[X] There are 0 failing Pester tests

## Changes this PR brings

- CIS test for symmetric encryption level of at least AES_128
- CIS test for asymmetric key size of at least 2048

Please add below the changes that this PR covers. It doesnt need an essay just the bullet points :-)

### CODE REVIEW

File: checks/Database.Tests.ps1 [hunk 1]
```diff
@@ -957,6 +957,48 @@ $ExcludedDatabases += $ExcludeDatabase
...
+                It "Can't Connect to $Psitem" {
+                    $true | Should -BeFalse -Because "The instance should be available to be connected to!"
+                }
+            }
+        }
+        else {
+            Context "Testing Symmetric Key Encruption Level at least AES_128 or higher on $psitem" {
+                @($InstanceSMO.Databases.Where{($(if ($Database) {$PsItem.Name -in $Database}else {$ExcludedDatabases -notcontains $PsItem.Name}))}).ForEach{
+                    It "$($psitem.Name) on $($psitem.Parent.Name) should have symmetric key encyption levels of at least AES_128" -Skip:$skip {
+                        $psitem.AutoClose | Should -BeTrue -Because "Symmetric Key Encruption Level at least AES_128 or higher for CIS compliance"
```
Comment (SQLDBAWithABeard) [inline checks/Database.Tests.ps1]: AutoClose property is not what you want I don't think

File: checks/Database.Tests.ps1 [hunk 2]
```diff
@@ -957,6 +957,48 @@ $ExcludedDatabases += $ExcludeDatabase
...
+                It "Can't Connect to $Psitem" {
+                    $true | Should -BeFalse -Because "The instance should be available to be connected to!"
+                }
+            }
+        }
+        else {
+            Context "Testing Asymmetric Key Size is 2048 or higher on $psitem" {
+                @($InstanceSMO.Databases.Where{($(if ($Database) {$PsItem.Name -in $Database}else {$ExcludedDatabases -notcontains $PsItem.Name}))}).ForEach{
+                    It "$($psitem.Name) on $($psitem.Parent.Name) asymmetric keys sizes of at least 2048" -Skip:$skip {
+                        $psitem.AutoClose | Should -BeTrue -Because "Asymmetric Key Size should be 2048 or higher for CIS compliance"
```
Comment (SQLDBAWithABeard) [inline checks/Database.Tests.ps1]: AutoClose property is not what you want I don't think

**Comment by tboggiano** [comment]
I lied this one was ready to go, I forgot I had just forgotten to push code up and you didn't actually ask for any changes....oops :)

**Comment by tboggiano** [comment]
Close 677
Close 678

**Pull request merged**